packages/gitbook/src/lib/cache/cache.ts

```diff
@@ -43,33 +43,38 @@ export interface CacheResult<Result> {
      * Time before ttl where the cache should be revalidated (in seconds).
      */
     revalidateBefore?: number;
+}
 
-    /**
-     * Tags to associate with the cache entry.
-     */
-    tags?: string[];
+export interface CacheDefinition<Args extends any[], Result> {
+    /** Unique name for the cache */
+    name: string;
+
+    /** Tag to associate to the entry */
+    tag?: (...args: Args) => string;
+
+    /** Filter the arguments that should be taken into consideration for the cache key */
+    getKeyArgs?: (args: Args) => any[];
+
+    /** Default ttl (in seconds) */
+    defaultTtl?: number;
+
+    /** When a request to the underlying resource will timeout. */
+    timeout?: number;
+
+    /** Function to get the value */
+    get: (...args: [...Args, CacheFunctionOptions]) => Promise<CacheResult<Result>>;
 }
 
 /**
  * Cache data from an async function.
  * We don't use the next.js cache because it has a 2MB limit.
  */
 export function cache<Args extends any[], Result>(
-    cacheName: string,
-    fn: (...args: [...Args, CacheFunctionOptions]) => Promise<CacheResult<Result>>,
-    options: {
-        /** Filter the arguments that should be taken into consideration for caching */
-        extractArgs?: (args: Args) => any[];
-
-        /** Default ttl (in seconds) */
-        defaultTtl?: number;
-
-        /** When a request to the underlying resource will timeout. */
-        timeout?: number;
-    } = {},
+    cacheDef: CacheDefinition<Args, Result>,
 ): CacheFunction<Args, Result> {
     // We stop everything after 10s to avoid pending requests
-    const timeout = options.timeout ?? 1000 * 10;
+    const timeout = cacheDef.timeout ?? 1000 * 10;
+    const defaultTtl = cacheDef.defaultTtl ?? 60 * 60 * 24;
```

Comment:
Might be worth mentioning that it's in seconds.

Reply:
it's documented in `defaultTtl`